node/packages/parser/src/ethereum/index.ts

```diff
@@ -52,21 +58,9 @@ export class TransactionParser {
     const sendAddress = tx.vin[0].addresses?.[0] ?? ''
     const receiveAddress = tx.vout[0].addresses?.[0] ?? ''
 
-    let result: ParseTxSpecific | undefined
-    switch (receiveAddress) {
-      case zrx.PROXY_CONTRACT: {
-        result = this.zrx.parse(tx)
-        break
-      }
-      case this.thor.routerContract: {
-        result = await this.thor.parse(tx)
-        break
-      }
-      case uniV2.ROUTER_CONTRACT: {
-        result = await this.uniV2.parse(tx)
-        break
-      }
-    }
+    // We expect only one Parser to return a result. If multiple do, we take the first and early exit.
+    const parser = await findAsyncSequential(this.parsers, async (parser) => !!(await parser.parse(tx)))
```

Comment:
It looks like this will run through all parsers, return on the first one that parsed the transaction, and then run `await parser?.parse(tx)` again. Can we just have the for of loop return the first result we get instead?

That function also seems general enough to have it as a top level parser helper instead of an ethereum helper specifically if we want to keep it as a helper function.